### ISSUE TITLE
Errors while creating new DSC pull client using configuration name

### ISSUE THREAD

**Issue by sameer-kumar** [PowerShell/PowerShell-Docs#578]
I'm trying to create a DSC pull client by following sample scripts provided in https://msdn.microsoft.com/en-us/powershell/dsc/pullclientconfignames?f=255&MSPPError=-2147217396

However, I get following errors:

```
The specified Property does not exist.
    + CategoryInfo          : MetadataError: (root/Microsoft/...gurationManager:String) [], CimException
    + FullyQualifiedErrorId : MI RESULT 12
    + PSComputerName        : localhost
```
 This is the script block I'm trying to execute:

```
[DSCLocalConfigurationManager()]
configuration PullClientConfigID
{
    Node localhost
    {
        Settings
        {
            RefreshMode = 'Pull'
            RefreshFrequencyMins = 30 
            RebootNodeIfNeeded = $true
        }

        ConfigurationRepositoryWeb DSCPullSrv
        {
            ServerURL = 'https://mypullserver:8080/PSDSCPullServer.svc'
            RegistrationKey = '5ecc0ec3-bc14-48ba-9651-cbf183e52a0f'
            ConfigurationNames = @('ClientConfig')
        }

        ResourceRepositoryWeb DSCResourceSrv
        {
            ServerURL = 'https://mypullserver:8080/PSDSCPullServer.svc'
            RegistrationKey = '5ecc0ec3-bc14-48ba-9651-cbf183e52a0f'
        }

        ReportServerWeb DSCReportSrv
        {
            ServerURL = 'https://mypullserver:8080/PSDSCPullServer.svc'
            RegistrationKey = '5ecc0ec3-bc14-48ba-9651-cbf183e52a0f'
        }
    }
}

PullClientConfigID -OutputPath 'C:\temp\PullClientConfigID\' 
Set-DSCLocalConfigurationManager localhost –Path 'C:\temp\PullClientConfigID\' –Verbose 
```

Am I missing something?

Thanks!

**Comment by ArieHein** [comment]
Remove the localhost for them Set-DSC command.
Or use -ComputerName localhost

**Comment by sameer-kumar** [comment]
I removed localhost from Set-Dsc command and now I'm thrown with

```
Registration of the Dsc Agent with the server https://mypullserver:8080/PSDSCPullServer.svc failed. The underlying error is: The attempt to register Dsc 
Agent with AgentId 22102E13-20FF-11E6-8115-005056A70AF0 with the server 
https://mypullserver:8080/PSDSCPullServer.svc/Nodes(AgentId='22102E13-20FF-11E6-8115-005056A70AF0') returned unexpected response code Forbidden. .
    + CategoryInfo          : InvalidResult: (root/Microsoft/...gurationManager:String) [], CimException
    + FullyQualifiedErrorId : RegisterDscAgentUnsuccessful,Microsoft.PowerShell.DesiredStateConfiguration.Commands.RegisterDscAgentCommand
    + PSComputerName        : localhost

```

I noticed that if I remove 'RegistrationKey' property from 'ResourceRepositoryWeb' and 'ReportServerWeb' section then it works. So, 'RegistrationKey' is provided just for 'ConfigurationRepositoryWeb' section.

Any ideas?

**Comment by ArieHein** [comment]
If you look closely at the sample in the link you posted, their case is 3 different pull servers, one for configurations, one for resources and one for reporting. When you have different pullservers, you need to use RegistrationKey in each, even if its value is the same,

In your script you do not need RegistrationKey for the Resource. In fact, youdont need that section at allif youre going to use the same pull server. Using ConfigurationRepositoryWeb is enough for the pull server to realise its also the resource provider.

As for ReportServer, have touse that section even if the URL is that same as the one for the ConfigurationRepositoryWeb . You just dont use a RegistrationKey at all.

So to sum it up:
- If you plan to use different pull servers for the various roles (config, resource, report) you need to use RegistraionKey (value is unimportant as long as it matches the keys you've set in the pull server creation script)
- If you want to use a ReportServer on the same node as your pullserver, you just need to have a section for it with the URL but no need for the RegistrationKey

**Comment by eslesar** [comment]
@sameer-kumar @ArieHein Thank you for the discussion. I'll close this now--it looks like questions have been answered. Feel free to re-open if there are further questions.
